R/stack.R

```diff
@@ -312,7 +312,7 @@ getSource <- function(frameId) {
     filename <- lineAndFile$filename
 
     if(is.null(lineAndFile$isFile)){
-      lineAndFile$isFile <- TRUE
+      lineAndFile$isFile <- is.character(filename) && !identical(filename, '')
```

Comment:
A simpler form of `!identical(filename, '')` is `nzchar(filename)`.

Reply:
I left the `identical(...)` in, because it is safer. E.g. `character(0)` would return `logical(0)`, which in turns causes errors when used in `if`.